src/librustdoc/clean/blanket_impl.rs

```diff
@@ -48,7 +48,7 @@ impl<'a, 'tcx> BlanketImplFinder<'a, 'tcx> {
                 // Require the type the impl is implemented on to match
                 // our type, and ignore the impl if there was a mismatch.
                 let Ok(eq_result) = infcx.at(&traits::ObligationCause::dummy(), param_env).eq(
-                    DefineOpaqueTypes::No,
+                    DefineOpaqueTypes::Yes,
```

Comment:
This is using a `DefiningAnchor::Error`, so this has no effect. leave a comment saying this.